addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -1021,17 +1035,32 @@ def reserve_pallet(self, pallet_name):
 
         self.ensure_one()
 
-        conflicting_batch = PickingBatch.search([
-            ('id', '!=', self.id),
+        domain = [
             ('state', '=', 'in_progress'),
-            ('u_last_reserved_pallet_name', '=', pallet_name),
-        ])
+        ]
+        if self.picking_ids.picking_type_id.u_reserve_pallet_per_picking:
+            domain.append(('u_reserved_pallets.pallet_name', '=', pallet_name))
+        else:
+            domain.extend([('id', '!=', self.id), ('u_last_reserved_pallet_name', '=', pallet_name)])
+        conflicting_batch = PickingBatch.search(domain)
         if conflicting_batch:
             raise ValidationError(
                 _("This pallet is already being used for batch %s.") %
                 conflicting_batch[0].name)
 
-        self.write({ 'u_last_reserved_pallet_name': pallet_name })
+        write_data = {'u_last_reserved_pallet_name': pallet_name}
+        if self.picking_ids.picking_type_id.u_reserve_pallet_per_picking:
```

Comment:
This will fail if picking_ids contains more than one picking.

Reply:
These have all been updated.